core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -326,6 +326,15 @@ public SystemSessionProperties(
                         QUERY_MAX_RUN_TIME,
                         "Maximum run time of a query (includes the queueing time)",
                         queryManagerConfig.getQueryMaxRunTime(),
+                        value -> {
+                            queryManagerConfig.getQueryMaxRunTimeHardLimit().ifPresent(hardLimit -> {
```

Comment:
QueryManagerConfig is mutable, should not be held on to

```java
queryManagerConfig.getQueryMaxRunTimeHardLimit()
                                .<Consumer<Duration>>map(hardLimit -> value -> {
                                    if (value.compareTo(hardLimit) > 0) {
                                        throw new TrinoException(
                                                INVALID_SESSION_PROPERTY,
                                                format("%s must not exceed 'query.max-run-time.hard-limit' %s: %s", QUERY_MAX_RUN_TIME, hardLimit, value));
                                    }
                                })
                                .orElse(value -> {}),
```

Reply:
Done